crates/bevy_ecs/src/schedule/schedule.rs

```diff
@@ -661,36 +657,159 @@ impl ScheduleGraph {
     }
 
     fn configure_sets(&mut self, sets: impl IntoSystemSetConfigs) {
-        let SystemSetConfigs { sets, chained } = sets.into_configs();
-        let mut set_iter = sets.into_iter();
-        if chained {
-            let Some(prev) = set_iter.next() else { return };
-            let mut prev_id = self.configure_set_inner(prev).unwrap();
-            for next in set_iter {
-                let next_id = self.configure_set_inner(next).unwrap();
-                self.dependency.graph.add_edge(prev_id, next_id, ());
-                prev_id = next_id;
+        self.configure_sets_inner(sets.into_configs(), false);
+    }
+
+    fn configure_sets_inner(
+        &mut self,
+        configs: SystemSetConfigs,
+        ancestor_chained: bool,
+    ) -> NodesResult {
+        match configs {
+            SystemSetConfigs::SystemSetConfig(config) => {
+                let node_id = self.configure_set_inner(config).unwrap();
+                if ancestor_chained {
+                    NodesResult {
+                        densely_chained: true,
+                        nodes: vec![node_id],
+                    }
+                } else {
+                    NodesResult {
+                        densely_chained: true,
```

Comment:
I don't understand why this is set to true here.

Reply:
This is set to true because an empty list of nodes fulfills the requirement of being densely chained, meaning that all nodes are linearly chained in the order they are defined. Since there are no nodes, this is trivially true.